compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -1142,7 +1151,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 // if x == 1 && y == 2 { .. }
                 //                 +
                 let actual_lhs_ty = self.check_expr(rhs_expr);
-                (Applicability::MaybeIncorrect, self.can_coerce(rhs_ty, actual_lhs_ty))
+                (
+                    Applicability::MaybeIncorrect,
+                    self.can_coerce(rhs_ty, actual_lhs_ty) | might_coerce(rhs_ty, actual_lhs_ty),
```

Comment:
binary `||`

```suggestion
                    self.can_coerce(rhs_ty, actual_lhs_ty) || might_coerce(rhs_ty, actual_lhs_ty),
```

Reply:
I've been making this mistake often lately, I wonder what's causing that 🤔 

Maybe that I write alternative patterns way more often than logical ors? No idea 🤷